src/Recognizer.ts

```diff
@@ -158,8 +158,8 @@ export abstract class Recognizer<TSymbol, ATNInterpreter extends ATNSimulator> {
 	 *
 	 * @since 4.3
 	 */
-	get parseInfo(): ParseInfo | undefined {
-		return undefined;
+	get parseInfo(): Promise<ParseInfo | undefined> {
```

Comment:
Might be clearer if we use async functions instead of explicit promises.

Reply:
I also think it might be a good convention to use an Async suffix on method names if they return promises.